WIP! Har 3008 muistutus suorasanktiona

Kauko opened the pull request.

Kauko committed:
Disabloi indeksin asetus jos laatupoikkeaman sanktio on muistutus

Kauko committed:
Suorasanktiolomakkeelle vaihtoehto syöttää muistutuksia

Kauko committed:
Palauta suorasanktioiden haussa tieto, onko sakko vai muistutus

Kauko committed:
Päivitä lomakkeen footer-fn dokumentaatiota

Kauko committed:
Anna tallentaa suorasanktio kun vaihtaa sanktiosta muistutukseen

Pitää käyttää lomakkeen footer-fn:ää, että validointi toimii
kunnolla.

Samalla tuli paljon formatointimuutoksia vahingossa.

Kauko commented:
Suosittelen katselmoimaan commiteittain. Yhdessä commitissa tuli lomakkeeseen liittyen formatointia vahingossa, joka voi vaan hämmentää.

Kauko committed:
Merge branch 'develop' into HAR-3008_muistutus_suorasanktiona

Kauko committed:
Merge branch 'develop' into HAR-3008_muistutus_suorasanktiona

Kauko committed:
Tallenna sanktioille luonti ja muokkaus metadata

Kauko committed:
Merge branch 'develop' into HAR-3008_muistutus_suorasanktiona

Kauko committed:
Poista sakko/muistutus valinta suorasanktiolomakkeesta

Kauko committed:
Poista sakko/muistutus valinta laatupoikkeamalomakkeesta

Kauko committed:
Älä lähetä palvelimelle sanktiotaulukon poistettuja rivejä

Kauko committed:
Lisää sanktioriville toimenpideinstanssi vain, jos se on sakko

Kauko committed:
Päivitä testidataa, muistutuksilla on sanktioryhma muistutus

Kauko committed:
Poista unohtunut debug-komponentti

Kauko committed:
Lomakkeen tallennusnapin muokkaus oli jäänyt puolitiehen

Kauko committed:
Migraatio: sanktion sakkoryhmä pakollinen, summattomat muistutuksia

Kauko committed:
Merge branch 'develop' into HAR-3008_muistutus_suorasanktiona

Kauko committed:
Merge branch 'develop' into HAR-3008_muistutus_suorasanktiona

Conflicts:
 tietokanta/src/main/resources/db/migration/V1_434__.sql

jarnovayrynen committed:
Pompauta migraationumeroa

tatut commented on tietokanta/src/main/resources/db/migration/V1_436__.sql:
```diff
@@ -0,0 +1,15 @@
+SET maara = NULL
+WHERE sakkoryhma = 'muistutus' :: sanktiolaji;
+
+DELETE FROM sanktio WHERE sakkoryhma IS NULL;
```
AIka rohkeaa. Onko tarkistettu paljonko on roskaa kannassa?

Kauko replied:
Jaa keskustelu ei toiminut ongelmitta kun tiedoston nimi muuttui. Näitä on tuotannossa 0, kuten kommentoin alla.

Kauko commented on tietokanta/src/main/resources/db/migration/V1_439__.sql:
```diff
@@ -0,0 +1,15 @@
+UPDATE sanktio
+SET sakkoryhma = 'muistutus' :: sanktiolaji
+WHERE maara IS NULL;
```
Näitä on tuotannossa 0.

Kauko commented on tietokanta/src/main/resources/db/migration/V1_439__.sql:
```diff
@@ -0,0 +1,15 @@
+
+UPDATE sanktio
+SET maara = NULL
+WHERE sakkoryhma = 'muistutus' :: sanktiolaji;
```
0 määrällä muistutuksia on 439
1300 määrällä 1
2000 määrällä 1

Kauko commented on tietokanta/src/main/resources/db/migration/V1_439__.sql:
```diff
@@ -0,0 +1,15 @@
+SET maara = NULL
+WHERE sakkoryhma = 'muistutus' :: sanktiolaji;
+
+DELETE FROM sanktio WHERE sakkoryhma IS NULL;
```
Näitä on tuotannossa 0

Kauko committed:
Merge branch 'develop' into HAR-3008_muistutus_suorasanktiona

Kauko committed:
Merge branch 'develop' into HAR-3008_muistutus_suorasanktiona

Kauko committed:
Pompauta migraatiota

jarnovayrynen committed:
Merge branch 'develop' into HAR-3008_muistutus_suorasanktiona

jarnovayrynen commented:
Liian pitkään auki, haara karkaa developista. Selvitettävä tämä asiakkaan kanssa onko tarve tällaiselle muutokselle.

jarnovayrynen committed:
Pompauta migraationumeroa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-3008_muistutus_suorasanktiona

jarnovayrynen committed:
Korjaa sanktioiden päivitys tallennuksen jälkeen

jarnovayrynen committed:
Mahdollista muistutus myös Sorat. ja Liik. ymp. hoitoon

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-3008_muistutus_suorasanktiona

jarnovayrynen committed:
Merge branch 'HAR-3008_muistutus_suorasanktiona' of github.com:finnishtransportagency/harja into HAR-3008_muistutus_suorasanktiona

jarnovayrynen committed:
Poista sanktio updatet migraatiosta

jarnovayrynen committed:
Palauta sanktio updatet migraatiosta

jarnovayrynen commented:
Äh, ei saa vieläkään mergettää. Jostain syystä kun päivittää sanktiota, luo yhä toisen laatupoikkeaman.. selvitettävä ennen mergeä.

EDIT: ilmeisesti kyseessä on laatupoikkeama, jolla on kaksi sanktiota.. todnäk ei siis blockeri

jarnovayrynen committed:
Käännä sanktiotyypin järjestys migraatiossa

jarnovayrynen committed:
Poista sanktiotyyppien päivitys, koska sanktiotyypit lisätty testidatana

jarnovayrynen committed:
Päivitä muistutus sanktiotyyppien testidataan

jarnovayrynen committed:
Vie LP:n sanktiogridiin LP koska validointi

jarnovayrynen committed:
Pompauta migraationumeroa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-3008_muistutus_suorasanktiona

jarnovayrynen committed:
Korjaa muistutuksen tunnistaminen sanktiosta

jarnovayrynen committed:
Korjaa parametrien järjestys sanktiota päivitettäessä

jarnovayrynen committed:
Korjaa muistutustarkistus

Muistutuksen totuudellisuus sijaitsee kontekstista riippuen joko kentässä laji tai
kentässä sakkoryhma. Domain-apufunktio tarkisti vain kentästä laji. Nyt lisätään
toinen funktio joka tarkistaa kentästä sakkoryhmä.

jarnovayrynen committed:
Poista duplikaatti require sanktio domainilta
